src/CMakeLists.txt

```diff
@@ -0,0 +1,1613 @@
+# linking
+
+add_executable(memory_utilities_unittest common/memory_utilities_unittest.F)
+add_executable(parallel_rng_types_unittest common/parallel_rng_types_unittest.F)
+add_executable(graph metadyn_tools/graph.F)
+add_executable(dumpdcd motion/dumpdcd.F)
+add_executable(xyz2dcd motion/xyz2dcd.F)
+add_executable(libcp2k_unittest start/libcp2k_unittest.c)
+
+set_target_properties(cp2k-bin PROPERTIES CXX_STANDARD 11)
+set_target_properties(cp2k-bin PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES
+                                          "CXX")
+set_target_properties(cp2k-bin PROPERTIES LINKER_LANGUAGE Fortran)
+set_target_properties(cp2k-bin PROPERTIES OUTPUT_NAME cp2k)
+target_link_libraries(cp2k-bin PUBLIC cp2k)
+
+foreach(_app ${__CP2K_APPS})
+  set_target_properties(${_app} PROPERTIES LINKER_LANGUAGE Fortran)
+  set_target_properties(${_app} PROPERTIES CXX_STANDARD 11)
+  set_target_properties(${_app} PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES
+                                           "CXX")
+  if(CP2K_USE_ACCEL MATCHES HIP)
+    set_target_properties(${__app} PROPERTIES HIP_ARCHITECTURES
+                                              ${CP2K_ACC_ARCH_NUMBER})
+  endif()
+  target_link_libraries(${_app} PUBLIC cp2k)
+endforeach()
+
+#
+# apps included in the grid module
+#
+
+list(APPEND __GRID_APPS "grid_miniapp" "grid_unittest")
+
+add_executable(
+  grid_miniapp
+  "grid/grid_miniapp.c;grid/grid_replay.c;${GRID_SRCS_C};${OFFLOAD_SRCS_C};${GRID_SRCS_GPU}"
+)
+add_executable(
+  grid_unittest
+  "grid/grid_unittest.c;grid/grid_replay.c;${GRID_SRCS_C};${OFFLOAD_SRCS_C};${GRID_SRCS_GPU}"
+)
+
+foreach(_app ${__GRID_APPS})
+  target_link_libraries(
+    ${_app}
+    PUBLIC $<$<BOOL:${CP2K_USE_CUDA}>:CUDA::cudart>
+           $<$<BOOL:${CP2K_USE_HIP}>:hip::device>
+           $<$<BOOL:${CP2K_USE_LIBXSMM}>:CP2K_LibXSMM::libxsmmf>
+           $<$<BOOL:${CP2K_USE_LIBXSMM}>:CP2K_LibXSMM::libxsmm>
+           CP2K_BLAS::blas
+           OpenMP::OpenMP_C
+           m)
+  set_target_properties(${_app} PROPERTIES LINKER_LANGUAGE C)
+  # target_compile_options(${_app} PUBLIC "${OpenMP_C_FLAGS}")
+  target_include_directories(
+    ${_app} PUBLIC $<$<BOOL:${CP2K_USE_CUDA}>:${CMAKE_CUDA_INCLUDE_DIRECTORIES}>
+                   $<$<BOOL:${CP2K_USE_HIP}>:${CMAKE_HIP_INCLUDE_DIRECTORIES}>)
+
+  if(CP2K_USE_CUDA)
+    set_target_properties(${_app} PROPERTIES CUDA_ARCHITECTURES
+                                             ${CP2K_ACC_ARCH_NUMBER})
+  endif()
+
+  # cmake 3.21 is needed for this
+  if(CP2K_USE_HIP)
+    set_target_properties(${_app} PROPERTIES HIP_ARCHITECTURES
+                                             ${CP2k_ACC_ARCH_NUMBER})
+  endif()
+endforeach()
+
+#
+# apps included in dbm
+#
+
+add_executable(dbm_miniapp dbm/dbm_miniapp.c ${DBM_SRCS_C} ${OFFLOAD_SRCS_C}
+                           ${DBM_SRCS_GPU})
+
+set_target_properties(dbm_miniapp PROPERTIES LINKER_LANGUAGE C)
+# target_compile_options(dbm_miniapp PUBLIC "${OpenMP_C_FLAGS}")
```

Comment:
Remove or uncomment? I think the `OpenMP::OpenMP_<lang>` target should anyway be enough for this.